lib/core/proxy.c

```diff
@@ -290,6 +292,15 @@ static void build_request(h2o_req_t *req, h2o_iovec_t *method, h2o_url_t *url, h
     }
 }
 
+static h2o_httpclient_t *detach_client(struct rp_generator_t *self)
+{
+    h2o_httpclient_t *client = self->client;
+    assert(client != NULL);
+    client->data = NULL;
```

Comment:
I believe that this being set to `NULL` introduces the possibility of a NULL deref in`proxy.c::on_head` https://github.com/h2o/h2o/pull/2010/files#diff-302e03bf5939f9d70777e2647bdb940aR444

Reply:
I take that back, i'm not so sure that this is the case if the following is applied:

```
@@ -587,7 +589,7 @@ static void proceed_request(h2o_httpclient_t *client, size_t written, h2o_send_s
     if (self == NULL) {
         return;
     }
-    if (!h2o_send_state_is_in_progress(send_state)) {
+    if (send_state == H2O_SEND_STATE_ERROR) {
         detach_client(self);
     }
     if (self->src_req->proceed_req != NULL)
```

Reply:
Thank you for the suggestion, you're right. I fixed the issue at https://github.com/h2o/h2o/pull/2010/commits/24d2036a5a9a3b7e218f8c82d3e92b5639760e65